### DISCUSSION TITLE
How to get row_key from row index?

### DISCUSSION THREAD

**Discussion by JOJ0** [Textualize/textual#5250] [Q&A] [answered]
Since I'm constantly struggling with DataTable basics, I'll ask upfront without endlessly searching (again) in existing docs and discussions:

How can I retrieve the row_key if I only know the index of the row (not sure if that term is correct, let's call it "when I know for example I want the row_key of the very first row (index 0?) in my currently displaying data table)

In the end I often would like to access the contents/value of the currently highlighted cell.

For example I want a key-press-event to execute an action, what is the basic code to get the cell contents of for example cell at column 3 and row 5:
```
    def action_toggle_sold(self):
        """When shortcut is pressed, toggle field "sold" in DB."""
        rlog = self.query_one(RichLog)
        col_key = "sold"  # This column label exists, and I successfully used it elsewhere in my program, so no problems there...
        row_key = ?  # I don't understand how to get this?
        rlog.write("row_key is", row_key)
```

PS: I (think) I understand by now, that `self.table.get_cell()` wants row_key and column_key as arguments and does not understand simple indexes (or whatever they are called) like `(3, 5)`

**Comment by JOJ0** [comment]
I'm very sorry but I was too dumb to search correctly. Is this my answer? https://github.com/Textualize/textual/discussions/3623

**Comment by TomJGooding** [comment]
It sound like you actually want [`DataTable.get_cell_at`](https://textual.textualize.io/widgets/data_table/#textual.widgets.DataTable.get_cell_at)?

**Reply by JOJ0** [reply]
Thanks a lot! Yes that works:

```
        row_num = self.table.cursor_row
        rlog.write(self.table.get_cell_at(Coordinate(row_num, 1)))
```

I just realized that my problem was that I was mixing row_keys and row ids/coordinates and that can't work. In the end I'm using the solution in the other link since I want to use the more readable column_key (eg. "sold" as in my example above).

Thanks again!

**Comment by recsater-uf** [comment, marked as answer]
Here is the actual answer to the title, not one modified based on author's original intention.

```python
def row_index_to_key(table, index): -> RowKey | None
    coordinate = Coordinate(index, 0)
     if table.is_valid_coordinate(coordinate):
        return table.coordinate_to_cell_key(coordinate).row_key
    return None
```
